### PR TITLE
Remove exceptions used for control flow

### PR DESCRIPTION
**Pull request by jfim**
This patch removes exceptions used for control flow when parsing
segment names.

### CODE REVIEW

File: pinot-common/src/main/java/com/linkedin/pinot/common/utils/SegmentName.java [hunk 1]
```diff
@@ -73,4 +74,30 @@ public int getPartitionId() {
...
     throw new RuntimeException("No partitionRange in " + getSegmentName());
   }
+
+  public static boolean isHighLevelConsumerSegmentName(String segmentName) {
+    if (segmentName.endsWith(SEPARATOR) || segmentName.startsWith(SEPARATOR)) {
+      return false;
+    }
+
+    String[] parts =  segmentName.split(SEPARATOR);
+    if (parts.length != 3 && parts.length != 5) {
```
Comment (mayankshriv) [inline pinot-common/src/main/java/com/linkedin/pinot/common/utils/SegmentName.java]: May be add a comment on how the HLC/LLC segments look like, so it is more clear on why 3/4/5?

**Comment by mayankshriv** [comment]
LGTM, except the minor nit mentioned.

File: pinot-common/src/main/java/com/linkedin/pinot/common/utils/SegmentName.java (hunk 1 above)
Comment (jfim) [inline pinot-common/src/main/java/com/linkedin/pinot/common/utils/SegmentName.java]: It's already documented in other parts (see https://github.com/linkedin/pinot/pull/252/files#diff-5aae5dfc010ab141a3c61f94cdebc42aR136 and https://github.com/linkedin/pinot/pull/252/files#diff-ae33b4a0fa4da0354102626cfcc98a0cR37) and most of this code should go away when we remove the high level consumer.

File: pinot-common/src/main/java/com/linkedin/pinot/common/utils/SegmentName.java [hunk 2]
```diff
@@ -73,4 +74,30 @@ public int getPartitionId() {
...
+
+    return true;
+  }
+
+  public static boolean isLowLevelConsumerSegmentName(String segmentName) {
+    if (segmentName.endsWith(SEPARATOR) || segmentName.startsWith(SEPARATOR)) {
+      return false;
+    }
+
+    String[] parts =  segmentName.split(SEPARATOR);
```
Comment (mcvsubbu) [inline pinot-common/src/main/java/com/linkedin/pinot/common/utils/SegmentName.java]: This logic is already in HLCSegmentName.  Constructing an HLCSegmentName (and catching an exception should be sufficient to decide whether it is HLC segment name or not, right? Let us not duplicate the parsing logic in multiple places.

**Pull request merged**